python/kserve/kserve/utils/numpy_codec.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2021 The KServe Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+
+
+def to_np_dtype(dtype):
```

Comment:
what would make this cleaner is if you had a map of np to dtype and you could loop through the map checking the type and converting it. that would be less repetitive than all the elifs

for example 

datatype_to_np = {"BOOL": bool, "INT8": np.int8}

 def to_np_dtype(dtype):
   for datatype, np in datatype_to_np.items():
     if dtype == datatype:
         return np 
         
         
 and u can use same map for the other way 
 def from_np_dtype(np_dtype):
   for datatype, np in datatype_to_np.items():
     if np_datatype == np:
          return datatype